Handle all supported browsers on Windows (#24)

GeorgeGkas opened the pull request:
Fixes #24

GeorgeGkas committed:
Handle all supported browsers on Windows (#24)

sindresorhus commented on windows.js:
```diff
@@ -4,8 +4,26 @@ import execall from 'execall';
+		browsers.push('Caption=\'msedge.exe\'')
+	}
+
+	const command = `wmic process where "${browsers.join(' or ')}" get CommandLine,ProcessId /format:list`;
```
It would be great if you could switch to `childProcess.execFile` while you're at it. That would simplify the quoting here.

GeorgeGkas replied:
Something like the following?? (It is a working example)

```js
const args = ['process', 'where', browsers.join(' or '), 'get','CommandLine,ProcessId', '/format:list'];
const stdout = await promisify(childProcess.execFile)('wmic', args, {maxBuffer: TEN_MEBIBYTE});
```

sindresorhus replied:
Yes

GeorgeGkas replied:
updated

GeorgeGkas committed:
Fix stylistic issues and tests

Update the default test for windows kill tabs function. The previous
version assumed no paraemter, but now we have to specify which
browser we want to kill.

GeorgeGkas committed:
Switch to execFile to simplify the quoting

GeorgeGkas committed:
Fix minor style issue

sindresorhus commented:
Make sure you manually test that it still works on Windows after these changes.

sindresorhus committed:
Update windows.js

GeorgeGkas committed:
Merge remote-tracking branch 'origin' into 24-support-browsers-windows

GeorgeGkas committed:
Merge branch '24-support-browsers-windows' of github.com:GeorgeGkas/kill-tabs into 24-support-browsers-windows

GeorgeGkas committed:
Handle edge case where no browser exists

In case the running machine do not have any browser installed,
the wmic command will report on stderr with
"No Instance(s) Available.". We don't handle this case as an
actual exception, so we do not throw. The script will never reach
fkill.

GeorgeGkas commented:
Added a commit to handle an edge case where the running machine do not have any browser installed, which will make the wmic command to report on stderr with "No Instance(s) Available.". We don't handle this case as an actual exception, so we do not throw.

GeorgeGkas commented:
Tests passed. If you are ok, we can resolve this PR.

sindresorhus commented on windows.js:
```diff
@@ -4,11 +4,29 @@ import execall from 'execall';
+		browsers.push('Caption=\'msedge.exe\'');
+	}
+
+	const args = ['process', 'where', browsers.join(' or '), 'get', 'CommandLine,ProcessId', '/format:list'];
```
You overwrote https://github.com/sindresorhus/kill-tabs/pull/25/commits/8849b97bf03847dfdf924f0111c5089e0cc5d4e5

GeorgeGkas replied:
oops I'm pushing a new commit to resolve it.

GeorgeGkas committed:
Restore changes made on 8849b97bf03847dfdf924f0111c5089e0cc5d4e5
